src/lib.rs

```diff
@@ -384,6 +384,14 @@ impl DeviceHandle {
 	}
 }
 
+fn to_cstring_safe(ptr: *const std::os::raw::c_char) -> CString {
+	if ptr == std::ptr::null() {
+		CString::new("").expect("CString::new failed")
```

Comment:
Use CString::default() get an empty string. It also allows us to avoid having to have the `expect`. 

```suggestion
		CString::default()
```